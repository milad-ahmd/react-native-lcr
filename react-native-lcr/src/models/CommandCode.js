/**
*   From LCP documentation:
*
*   Command Codes
*   Command   Description
*         0   Start/Resume a delivery.
*         1   Pause the current delivery.
*         2   End the current delivery and print the ticket if the settings allow.
*         3   Place the LCR in auxiliary state.
*         4   Place the LCR in shift state and print the ticket if settings allow.
*         5   Place the LCR in calibration state. This only applies if the LCR was previously in factory mode.
*         6   Print a ticket based on the current state of the LCR. This command does not end
*               any deliveries, nor will it print a ticket if a delivery is active.
*/

const CommandCode = {
  StartResume: 0,
  Pause: 1,
  Print: 2,
  ShiftPrint: 2,
}

export default CommandCode
