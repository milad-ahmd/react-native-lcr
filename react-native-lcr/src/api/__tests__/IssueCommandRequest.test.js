import issueCommandRequest from "../../../app/api/requests/IssueCommandRequest"
import CommandCode from "../../../app/models/CommandCode"

describe("setFieldDataRequest", () => {
  it("requests the field data", () => {
    let command = CommandCode.EndAndPrintDelivery
    let toNode = 1

    const result = issueCommandRequest(toNode, command)
    expect(result).toMatchSnapshot()
  })
})
