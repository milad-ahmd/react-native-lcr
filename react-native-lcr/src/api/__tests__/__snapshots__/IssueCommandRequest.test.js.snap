// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setFieldDataRequest requests the field data 1`] = `
Object {
  "address": 1,
  "body": Array [
    36,
    undefined,
  ],
  "requestName": "issue command: undefined",
  "response": StreamResponse {
    "data": Object {
      "code": DecimalType {
        "length": 1,
      },
      "devStatus": DecimalType {
        "length": 1,
      },
    },
  },
}
`;
